### PR TITLE
First version

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
@EPashkin: I'm wondering: why all functions using Gtk/Gdk/cairo widgets aren't generated in here? Did I miss something?

### CODE REVIEW

**Comment by EPashkin** [comment]
No `manual` from gtk?

**Comment by GuillaumeGomez** [comment]
What do you mean?

File: src/auto/completion.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        }
+    }
+
+    fn hide(&self) {
+        unsafe {
+            ffi::gtk_source_completion_hide(self.to_glib_none().0);
+        }
+    }
+
+    //fn move_window(&self, iter: /*Ignored*/&mut gtk::TextIter) {
```
Comment (EPashkin) [inline src/auto/completion.rs]: Here example, you need add "Gtk.TextIter" to manual array

Comment (GuillaumeGomez) [inline src/auto/completion.rs]: Oh I see! Thanks!

**Comment by EPashkin** [comment]
`target` folder added, also libs normally don't contains lock files

**Comment by GuillaumeGomez** [comment]
I need to clean everything up anyway. Starting with `Cargo.toml` which points to invalid dependencies.

**Comment by GuillaumeGomez** [comment]
@EPashkin: I think it's done here. Once I have your confirmation, I'll move this repository to `gtk-rs` and push the crates to crates.io and then merge this PR (I still need to update dependencies...).

File: src/completion.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+impl Completion {
+    pub fn create_context<'a, P: Into<Option<&'a mut gtk::TextIter>>>(&self, position: P) -> Option<CompletionContext> {
+        let mut position = position.into();
+        let position = position.to_glib_none_mut();
+        unsafe {
+            from_glib_none(ffi::gtk_source_completion_create_context(self.to_glib_none().0, position.0))
+        }
+    }
+}
```
Comment (EPashkin) [inline src/completion.rs]: Empty line

File: src/gutter.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+use Gutter;
+
+impl Gutter {
+    pub fn get_padding(&self, xpad: &mut i32, ypad: &mut i32) {
+        unsafe {
+            ffi::gtk_source_gutter_get_padding(self.to_glib_none().0, xpad, ypad);
+        }
+    }
+}
```
Comment (EPashkin) [inline src/gutter.rs]: Empty line

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/gutter.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib::translate::ToGlibPtr;
+
+use Gutter;
+
+impl Gutter {
+    pub fn get_padding(&self, xpad: &mut i32, ypad: &mut i32) {
```
Comment (EPashkin) [inline src/gutter.rs]: IMHO it better be get_padding(&self) -> (i32, i32).

**Comment by EPashkin** [comment]
Maybe need add manual Gtk.TextMark, ScrollStep
Strange StyleSchemeChooserWidget commented constructor.
But other is normal.

**Comment by EPashkin** [comment]
So IMHO it can merged

**Comment by GuillaumeGomez** [comment]
Ok, so first I merge this PR, then I create the crate branch to push to crates.io, then I push to crates.io, then I push the changes (on both crate and master branches) and then I put it into the `gtk-rs` organization.

**Pull request merged**